pkg/util/util.go

```diff
@@ -430,13 +431,14 @@ func ResolvePVCFromPod(pod *corev1.Pod, pvcLister corelisterv1.PersistentVolumeC
 			pvc, err := pvcLister.PersistentVolumeClaims(pod.Namespace).Get(pvcName)
 			if err != nil {
 				klog.Errorf("Get PVC %s/%s error: %v", pod.Namespace, pvcName, err)
-				continue
+				return nil, err
 			}
 			pvcs = append(pvcs, pvc)
 		}
 	}
 	if len(pvcs) == 0 {
-		return nil, errors.NewNotFound(corev1.Resource("pvc"), pod.Name)
+		klog.Errorf("no pvc found for pod %s", pod.Name)
+		return nil, errors.NewNotFound(corev1.Resource("pvc"), "")
```

Comment:
The error message is not clear enough, what about just return new error with the log message in the last line.

Reply:
maybe the caller will check error type with `errors.IsNotFound()` and skip a `NotFound` error?

Reply:
The current callers do not check the error type now and if some do check later, we can define a new NotFound error with a clearer message, the current one will just report no PVC found, without any useful info, e.g., which PVC, for which Pod in which ns, right?

Reply:
PD has [ut case](https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/pd_failover_test.go#L323) that asserts no pvc found but deleting pod succeeded.
Fixed in https://github.com/pingcap/tidb-operator/pull/3820/files#diff-0fa449830b5dc2b274b7c95a58b5071edc23f1e44af84a99c1fa9e600767829fR440